tools/IceRpc.BuildTelemetry.Reporter/SystemHelpers.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ZeroC, Inc.
+
+using System.Runtime.InteropServices;
+
+namespace IceRpc.BuildTelemetry;
+
+/// <summary>
+///  Helper class to get information about the operating system.
+/// </summary>
+internal static class SystemHelpers
+{
+    /// <summary> Get the operating system name in a human-readable format.</summary>
+    /// <returns> The operating system name.</returns>
+    internal static string GetOperatingSystem()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return "Windows";
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            return "macOS";
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            return "Linux";
+        }
+        else
+        {
+            return "Unknown OS";
```

Comment:
Maybe return `RuntimeInformation.OSDescription` here, or maybe we should collect both always.

Reply:
The `RuntimeInformation.OSDescription` is too high of cardinality:

```
Darwin 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:14:59 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8122
```

We would never really be able to use of filter or anything based on this as every single telemetry data sent would be different.

Reply:
It contains a timestamp, if we were to store it we would have to strip some of that data out of it.

Reply:
As far as I know, the timestamp is for the OS build, there are not that many OS builds.